enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,482 @@
+with additional workers recovers after an upgrade by running conformance tests
+post-upgrade.
+
+TODO: Describe day-1 tests?
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+In the non-goals section it's mentioned that this enhancement does not apply to
+clusters which have been installed prior to the enhancement, so their upgrade
+is not discussed.
+
+This enhancement, to the best of my knowledge, should have no problems
+persisting across cluster upgrades. The Test Plan section describes how this
+will be tested.
+
+### Version Skew Strategy
+
+Does not apply, to the best of my understanding.
+
+### Operational Aspects of API Extensions
+
+This enhancement does not modify/add any API
+
+#### Failure Modes
+
+This enhancement does not modify/add any API
+
+#### Support Procedures
+
+This enhancement does not modify/add any API
+
+## Implementation History
+
+Not yet applicable
+
+## Drawbacks
+
+- The pinning of the `IngressController` to the "master" pool is another change
+which would make single-node clusters slightly different from multi-node
+clusters, and any such difference is naturally not ideal.
+
+- TBD
+
+## Alternatives
+
+- If just one worker is needed, require user to add yet another worker so they
+could form a compact 3-node cluster where all nodes are both workers and
+control-plane nodes. This kind of topology is already supported by OCP. This
+will avoid the need for OCP to support yet another topology. It has the obvious
+downside of requiring a "useless" node the user didn't really need. It also
+means the user now has to run more control-plane workloads to facilitate HA -
+for example, 2 extra replicas of the API server which consume a lot of memory
+resources. From an engineering perspective, it would require us to make the
+"Control-plane Topology" parameter dynamic and make sure all operators know to
+react to changes in that parameter (it will have to change from "SingleReplica"
+to "HighlyAvailable" once those 2 new control-plane nodes join the cluster). I
+am not aware of the other engineering difficulties we would encounter when
+attempting to support the expansion of a single-node control-plane into a
+three-node control-plane, but I think they would not be trivial.
+
+- Adjust the "baremetal" platform to support single-node installations and make
+users and the Assisted-Installer use that platform instead of the "none"
+platform for single control-plane node cluster installations. The baremetal
+platform solves the issue described in this enhancement with virtual IP
+addresses/keepalived. This approach was dismissed due to much higher
+development efforts and additional processes that would need to run on the
+already resource constrained single control-plane node. Furthermore, even once
```

Comment:
Which additional processes would be running? What other implementation work is needed?

Reply:
I assume @omertuc meant keepalived, haproxy, .... all the on prem networking.